src/wp-includes/rest-api/endpoints/class-wp-rest-users-controller.php

```diff
@@ -717,15 +717,18 @@ public function update_item( $request ) {
 
 		$id = $user->ID;
 
-		if ( ! $user ) {
+		if ( ! $id ) {
```

Comment:
If my understanding is correct, this condition can be removed as `$user->ID` can never be falsy. Why?

- This method calls `WP_REST_Users_Controller::get_user()`.
- which calls `get_userdata()`.
- which calls `get_user_by()`.
- which calls `WP_User::get_data_by()`.
- which has [this condition](https://github.com/WordPress/wordpress-develop/blob/6.0.2/src/wp-includes/class-wp-user.php#L207-L209):

```php
if ( 'id' === $field ) {
	//...
	$value = (int) $value;
	if ( $value < 1 ) {
		return false;
	}
	// ...
```
- `false` goes back to `get_user_by()`:
```php
if ( ! $userdata ) {
	return false;
}
```
- `false` goes back to `get_userdata()`:
```php
return get_user_by( 'id', $user_id );
```
- `false` goes back to `WP_REST_Users_Controller::get_user()`:
```php
if ( empty( $user ) || ! $user->exists() ) {
	return $error;
}
```
- `empty( false ) === true` => returns `$error`.
- `$error` (`WP_Error`) to `WP_REST_Users_Controller::update_item()` (this method):
```php
if ( is_wp_error( $user ) ) {
	return $user;
}
```
- The method returns a `WP_Error` object before `$id = $user->ID` is hit.

Reply:
Exactly ;-)

Reply:
I opened a new Trac ticket for this bug fix https://core.trac.wordpress.org/ticket/56662 and added the historical context for backtracing how it happened and why this code exists. IMO this conditional branch can be safely removed.

Reply:
@hellofromtonya Shall I update the second commit to that effect ? Or are you already making that change ?

Reply:
@jrfnl If you have time, sure. Might be good to create a PR and cherry pick this commit as well as add the removal commit. That gives 2 separate options for gaining consensus. What do you think?

Reply:
Done: https://github.com/WordPress/wordpress-develop/pull/3344